services/service-core/src/auth/test/keycloak.service.spec.ts

```diff
@@ -80,21 +80,37 @@ describe('KeycloakService', () => {
     });
 
     it('throws an error when it fails to fetch', async () => {
-      const mockError = new Error('Error: Fetch error');
+      const mockFailedResponse = new Response(null, {
+        status: 400,
+        statusText: 'Bad Request'
+      });
+
+      jest.spyOn(global, 'fetch').mockResolvedValue(mockFailedResponse);
+
+      await expect(keycloakService.getAdminToken()).rejects.toThrowError(
+        'Bad Request'
+      );
+    });
+
+    it('throws an error with the status text when fetch is not OK', async () => {
+      const mockErrorResponse = new Response(null, {
+        status: 500,
+        statusText: 'Internal Server Error'
+      });
 
-      jest.spyOn(global, 'fetch').mockRejectedValue(mockError);
+      jest.spyOn(global, 'fetch').mockResolvedValue(mockErrorResponse);
 
       await expect(keycloakService.getAdminToken()).rejects.toThrowError(
-        mockError
+        'Internal Server Error'
```

Comment:
Since you're spying for a mocked response, it is better to make error message clear in this regard, like `Mocked Error Message`. Upon reading code of the test it will help to understand it